model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala

```diff
@@ -216,22 +100,60 @@ case class SpectroscopyModeRow(
   def intervalCenter(cw: Wavelength): Option[CentralWavelength] =
     ModeCommonWavelengths
       .wavelengthInterval(cw)(this)
-      .map(interval =>
+      .map: interval =>
         interval.lower.pm.value.value + (interval.upper.pm.value.value - interval.lower.pm.value.value) / 2
-      )
       .flatMap(pms => Wavelength.fromIntPicometers(pms))
       .map(CentralWavelength(_))
+
+  import lucuma.core.model.sequence.gmos.longslit.DefaultRoi
+
+  def withModeOverridesFor(
+    wavelength:   Wavelength,
+    profiles:     NonEmptyList[SourceProfile],
+    imageQuality: ImageQuality
+  ): Option[SpectroscopyModeRow] =
+    intervalCenter(wavelength).flatMap: cw =>
+      val instrumentConfig: Option[InstrumentConfig] =
+        instrument.instrument match
+          case Instrument.GmosNorth | Instrument.GmosSouth =>
+            instrument match
+              case i @ InstrumentConfig.GmosNorthSpectroscopy(grating, fpu, _, None) =>
+                i.copy(modeOverrides =
+                  InstrumentOverrides
+                    .GmosSpectroscopy(
+                      cw,
+                      GmosCcdMode.defaultGmosNorth(profiles, fpu, grating, imageQuality),
```

Comment:
Nice, I didn't notice this method. Is it from `lucuma-core`?

Reply:
It's an extension I created. But I think it should be moved to `lucuma-core` in the future.